pkg/scheduler/objects/application.go

```diff
@@ -107,6 +111,48 @@ type Application struct {
 	sync.RWMutex
 }
 
+type ApplicationSummary struct {
+	ApplicationID string
+	SubmissionTime time.Time
+	StartTime time.Time
+	FinishTime time.Time
+	User string
+	Queue string
+	State string
+	RmID string
+	ResourceUsage* resources.UsedResource
+}
+
+func (as *ApplicationSummary) DoLogging() {
+	log.Logger().Info("YK_APP_SUMMARY:",
+		zap.String("appID", as.ApplicationID),
+		zap.Int64("submissionTime", as.SubmissionTime.UnixMilli()),
+		zap.Int64("startTime", as.StartTime.UnixMilli()),
+		zap.Int64("finishTime", as.FinishTime.UnixMilli()),
+		zap.String("user", as.User),
+		zap.String("queue", as.Queue),
+		zap.String("state", as.State),
+		zap.String("rmID", as.RmID),
+		zap.Any("resourceUsage", as.ResourceUsage.UsedResourceMap))
+}
+
+func (sa *Application) GetApplicationSummary(rmID string) *ApplicationSummary {
+	sa.RLock()
+	defer sa.RUnlock()
+	appSummary := &ApplicationSummary{
+		ApplicationID: sa.ApplicationID,
+		SubmissionTime: sa.SubmissionTime,
+		StartTime: sa.startTime,
+		FinishTime: sa.finishedTime,
+		User: sa.user.User,
+		Queue: sa.queuePath,
+		State: sa.stateMachine.Current(),
```

Comment:
This is guarded by a read lock inside the state machine. It's most likely not an issue, but it's best not to have multiple locks grabbed this way. So move this out before `sa.Lock()` and save it to a variable.